packages/assets-controllers/src/TokenRatesController.ts

```diff
@@ -540,4 +566,60 @@ export class TokenRatesController extends PollingControllerV1<
   }
 }
 
+/**
+ * A deferred Promise.
+ *
+ * A deferred Promise is one that can be resolved or rejected independently of
+ * the Promise construction.
+ */
+type DeferredPromise = {
+  /**
+   * The Promise that has been deferred.
+   */
+  promise: Promise<void>;
+  /**
+   * A function that resolves the Promise.
+   */
+  resolve: () => void;
+  /**
+   * A function that rejects the Promise.
+   */
+  reject: (error: unknown) => void;
+};
+
+/**
+ * Create a defered Promise.
+ *
+ * TODO: Migrate this to utils
+ *
+ * @param args - The arguments.
+ * @param args.suppressUnhandledRejection - This option adds an empty error handler
+ * to the Promise to suppress the UnhandledPromiseRejection error. This can be
+ * useful if the deferred Promise is sometimes intentionally not used.
+ * @returns A deferred Promise.
+ */
+function deferredPromise({
```

Comment:
Nit: Maybe this should be called `createDeferredPromise`?

Reply:
Agreed, updated in https://github.com/MetaMask/core/pull/3635/commits/d4e77d11af40db1b6b58056b3687ceca26c494f6